language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverObjectResolver.java

```diff
@@ -164,9 +164,14 @@ private Hover getHoverObjectForSymbol(FunctionSymbol functionSymbol) {
                 }
                 String paramName = param.getName().get();
                 String desc = paramsMap.get(paramName);
+                Optional<String> defaultValueForParam = CommonUtil.getDefaultValueForType(param.typeDescriptor());
```

Comment:
This is incorrect. What we have done is getting the default value for the parameter's type. Instead we should show the default value of defaultable parameters. Here we should first check if `param` is a defaultable param (check `ParameterKind`). Then, we need to find out it's default value (using the `DefaultableParameterNode.expression()` or via the type symbol)